File: src/dataset/python/isbn-verifier/gemini/GeminiFlash_isbn-verifier_v1.py
def is_valid(isbn):
    isbn = isbn.replace('-', '')
    if len(isbn) != 10:
        return False

    sum_isbn = 0
    x = 10
    for i, char in enumerate(isbn):
        if i < 9:
            if not char.isdigit():
                return False
            sum_isbn += int(char) * x
        else:
            if char == 'X':
                sum_isbn += 10
            elif char.isdigit():
                sum_isbn += int(char)
            else:
                return False
        x -= 1

    return sum_isbn % 11 == 0

File: src/dataset/typescript/isbn-verifier_Exercism-typescript-shybyte/gemini/GeminiFlash_isbn-verifier_v1.ts
export default class ISBN {
    private readonly isbnString: string;

    public constructor(isbnString: string) {
        this.isbnString = isbnString;
    }

    public isValid(): boolean {
        let cleanedIsbn = this.isbnString.replace(/-/g, '');
        const length = cleanedIsbn.length;

        if (length === 0) {
            return false;
        }

        let sum = 0;
        for (let i = 0; i < length; i++) {
            const char = cleanedIsbn[i];
            let digit: number;

            if (char === 'X' && i === length - 1) {
                digit = 10;
            } else if (char >= '0' && char <= '9') {
                digit = char.charCodeAt(0) - '0'.charCodeAt(0);
            } else {
                return false;
            }

            sum += digit * (10 - i);
        }

        return sum % 11 === 0;
    }
}